File: scratch/recover.py
import codecs

with open('temp_DashboardScreen.txt', 'r', encoding='utf-8') as f:
    content = f.read().strip()

if content.startswith('"'):
    content = content[1:]
if content.endswith('"'):
    content = content[:-1]

# Unescape using codecs
content = codecs.decode(content, 'unicode_escape')

with open('src/screens/DashboardScreen.tsx', 'w', encoding='utf-8') as out:
    out.write(content)
    
print("Recovery completed perfectly")

File: scratch/recover.js
const fs = require('fs');

try {
  let data = fs.readFileSync('temp_DashboardScreen.txt', 'utf8');
  // It seems to be a JSON string literal (starting and ending with quotes, with \n escaped)
  if (data.startsWith('"') && data.endsWith('"\n')) {
      data = JSON.parse(data.trim());
  } else if (data.startsWith('"') && data.endsWith('"')) {
      data = JSON.parse(data);
  } else if (data.startsWith('"')) {
      // Sometimes it might not have the exact closing quote at the end due to whitespace
      data = JSON.parse(data.trim());
  }
  
  fs.writeFileSync('src/screens/DashboardScreen.tsx', data);
  console.log('Successfully recovered DashboardScreen.tsx');
} catch(e) {
  console.error('Failed to parse:', e);
}
